### PR TITLE
Test and fix QgsRectangle::isNull

### PR DESCRIPTION
**Pull request by strk**
Adds tests for QgsRectangle::isNull, improve toString to show the difference between Null and Empty, eventually fix the 0,0,0, rectangle case to still be Empty and NOT Null (to match documentation)

### CODE REVIEW

File: src/core/geometry/qgsrectangle.h [hunk 1]
```diff
@@ -43,7 +43,12 @@ class CORE_EXPORT QgsRectangle
   public:
 
     //! Constructor for a null rectangle
-    QgsRectangle() = default; // optimised constructor for null rectangle - no need to call normalize here
+    QgsRectangle()
+      : mXmin( std::numeric_limits<double>::max() )
+      , mYmin( std::numeric_limits<double>::max() )
+      , mXmax( -std::numeric_limits<double>::max() )
+      , mYmax( -std::numeric_limits<double>::max() )
```
Comment (nyalldawson) [inline src/core/geometry/qgsrectangle.h:50]: I'm wondering if it's cleaner to just set these all as nan, and adjust the isnull test accordingly. What's your thoughts?

**Comment by nyalldawson** [comment]
I think this is heading in the right direction, nice work @strk !

**Comment by strk** [comment]
Reopening this PR didn't work, so I'm creating a new one replacing this

**Comment by strk** [comment]
Superceeded by #47404

**Pull request closed without merging**